src/compiler/parser.ts

```diff
@@ -6259,7 +6259,7 @@ namespace ts {
                 }
 
                 function parseJSDocIdentifier(): Identifier {
-                    if (token !== SyntaxKind.Identifier) {
+                    if (token !== SyntaxKind.Identifier && !isTSOnlyKeyword(token)) {

```

Comment:
well.. not sure i see much value..

`/** param {myNonExistingTypeName} x **/` is not an error. so why is it different from 
`/** param {import} x **/`